docs/extending/hooks.md

```diff
@@ -55,6 +55,46 @@ function custom_request_details( array $request_details, HttpQueryInterface $que
 add_filter( 'remote_data_blocks_request_details', 'custom_request_details', 10, 3 );
 ```
 
+### remote_data_blocks_query_input_variables
+
+Filter the query input variables prior to query execution. This filter is useful for modifying the input variables for the current page-load, e.g., by pulling in data from query variables or other context. See [Overrides](overrides.md) for more information.
+
+```php
+add_filter( 'remote_data_blocks_query_input_variables', function ( array $input_variables, array $enabled_overrides, string $block_name, array $block_context ): array {
+	if ( true === in_array( 'my_override', $enabled_overrides, true ) ) {
+		$override_value = get_query_var( 'override_id' );
+
+		if ( ! empty( $override_value ) ) {
+			$input_variables['id'] = $override_value;
+		}
+	}
+
+	return $input_variables;
+}, 10, 4 );
+```
+
+Note that if your filter modify a query's input variables, it will modify the object cache key for that query execution, possibly resulting in a cache miss.
```

Comment:
This had a grammar issue, so I made a fix + some more tweaks. Feel free to just fix the grammar issue though if you want and undo my other suggestions.
```suggestion
Keep in mind modifying query input variables will affect the object cache key for that query execution. This could result in a cache miss.
```